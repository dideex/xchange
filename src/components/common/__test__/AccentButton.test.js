import React from 'react'
import Component from '../AccentButton'
import {shallow} from 'enzyme'

describe('Accent button behaviour', () => {
  const toggle = jest.fn()
  const wrapper = shallow(<Component toggle={toggle}>Button</Component>)

  it('Markup', () => {
    const snapshotWrapper = shallow(
      <Component toggle={() => {}}>
        <div>Snapshot</div>
      </Component>,
    )
    expect(snapshotWrapper.html()).toMatchSnapshot()
  })

  it('Children shoul be', () => {
    expect(wrapper.html()).toContain('Button')
  })

  it('Toggle button should be', () => {
    wrapper.simulate('click')
    expect(toggle).toHaveBeenCalledTimes(1)
  })
})
