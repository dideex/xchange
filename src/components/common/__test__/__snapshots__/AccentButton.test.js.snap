// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accent button behaviour Markup 1`] = `"<button class=\\"css-19o48nz\\"><div>Snapshot</div></button>"`;
